src/main/java/com/google/cloudassets/discovery/AssetKind.java

```diff
@@ -24,5 +35,50 @@ private AssetKind(String kind) {
     public String toString() {
         return this.kindString;
     }
+
+    /**
+     * @return a string representing the table name of a given asset kind.
+     */
+    public String getAssetTableName() {
+        String errorMsg = " was configured as the asset table in Asset_Tables_Config for the '"
+                + this.kindString + " asset kind. Please make sure that exactly one table is configured"
+                + " for each asset kind.";
+        String queryStr = "SELECT DISTINCT assetTableName FROM Asset_Tables_Config WHERE assetKind = '"
+                + this.kindString + "'";
+        return getTableName(queryStr, errorMsg);
+    }
+
+    /**
+     * @return a string representing the main asset table name.
+     */
+    public static String getMainTableName() {
+        String errorMsg = " was configured as the main table in Asset_Tables_Config. "
+                + "Please make sure that exactly one table has the 'isMainTable' flag on.";
+        String queryStr = "SELECT DISTINCT assetTableName FROM Asset_Tables_Config WHERE isMainTable = True";
+        return getTableName(queryStr, errorMsg);
+    }
+
+    /*
+    This function returns a table name returned from the provided queryStr. If there is not exactly
+    one table name returned from the provided query it means that the Asset_Tables_Config is not
+    configured properly and therefore unexpected behaviors may arise. In this case an error is logged.
+     */
+    private static String getTableName(String queryStr, String errorMsg) {
+        ResultSet resultSet = executeStringQuery(queryStr);
+        String tableName = null;
+        if (resultSet.next()) {
+            tableName = resultSet.getString("assetTableName");
+        } else {
+            String error_msg = "No table" + errorMsg;
```

Comment:
Few comments about this:
- 'error_msg' is not according to the style guide
- Having both 'error_msg' and 'errorMsg' is very confusing.
- Why not throw an exception? This failure is probably fatal.
- Instead of passing in a partial error-message and putting it together here, it is better practice to throw and exception here, and re-throw in the caller function, wrapping this exception in another one adding more info (e.g. adding the kind)

Reply:
I was indeed wondering what was the best way to handle these kinds of errors:
1. The getAssetTableName & getMainTableName functions which are using this function are being called in several places in our program so throwing an error will make it necessary to handle these exceptions in the same way in different places. In order to avoid this I thought to run a validation function on our configuration tables once in the beginning of the back-end process, but this might be redundant.. WDYT?
2. Should we stop the program execution upon encountering this error? If so, what is the best way to do this?
3. Should I create a new Exception class for these errors or is this redundant?

Reply:
These are indeed non-trivial questions... Let's go over these questions over VC

Reply:
Fixed this issue with new exceptions as we discussed (as a first step before adding system tests). 
Would love to get your input if it looks okay now.